contributors/design-proposals/apiserver-count-fix.md

```diff
@@ -49,75 +50,27 @@ Endpoints would never become consistent.
 
 | Kubernetes Release  | Quality | Description |
 | ------------- | ------------- | ----------- |
-| 1.9           | alpha         | <ul><li>Add a new reconciler</li><li>Add a command-line type `--alpha-apiserver-endpoint-reconciler-type`<ul><li>configmap</li><li>default</li></ul></li></ul>
-| 1.10          | beta          | <ul><li>Turn on the `configmap` type by default</li></ul>
+| 1.9           | alpha         | <ul><li>Add a new reconciler</li><li>Add a command-line type `--alpha-apiserver-endpoint-reconciler-type`<ul><li>storage</li><li>default</li></ul></li></ul>
+| 1.10          | beta          | <ul><li>Turn on the `storage` type by default</li></ul>
 | 1.11          | stable        | <ul><li>Remove code for old reconciler</li><li>Remove --apiserver-count</li></ul>
 
-The MasterCountEndpointReconciler does not meet the current needs for
-durability of API Endpoint creation, deletion, or failure cases.
-
-Create a new `MasterEndpointConfigMapReconciler` within
-master/controller.go.
-
-Add a `kube-apiserver-config` ConfigMap in the `kube-system`
-namespace. The duration found within the Coordination map would be configurable by
-admins without a recompile. The ConfigMap would include the following:
-
-```go
-KubeApiServerConfigMap{
-  "expiration-duration":        "1m",
-  ... [other flags potentially]
-}
-```
-
-The Coordination ConfigMap would be formed such that:
-
-```go
-CoordinationConfigMap{
-  "[namespace]/[name]": "[KubeAPIServerEndpoint]",
-}
-```
-
-*Key Format*: ip-[IP String Formatted]-[port]
-
-```go
-KubeAPIServerEndpointConfigMap{
-  "master-count":                "0",
-	"ip-2001-4860-4860--8888-443": "time.Time",
-	"ip-192-168-0-3-443":          "time.Time",
-}
-```
-
-***TODO: should it a serialized struct, or the raw time value?***
-
-The reconcile loop will expire endpoints that do not meet the duration.
-On each reconcile loop (the loop runs every 10 seconds currently,
-but interval will be changed to 80% of the `expiration-duration`):
-
-1. GET `kube-apiserver-endpoints` ConfigMap (as endpointMap)
-1. Update the timestamp for the currently running API server
-1. Remove all endpoints where the timestamp is greater than `expiration-duration` from the configMap.
-1. Do a GET on the `kube-apiserver-endpoints`
-1. Construct a PATCH operation with the deleted endpoints
+The MasterCountEndpointReconciler does not meet the current needs for durability
+of API Endpoint creation, deletion, or failure cases.
 
-configmap.yml:
+Custom Resource Definitions and ConfigMaps were proposed, but since they are
```

Comment:
I can understand ConfigMap has the watching issue. But CRD should not be watched globally. The problem with CRD , as far as I can tell,  is the not super clean layering, and potentially we might want something different that can handle both liveness and locking as a k8s native resource. And we are not sure yet.